app/docs/0.14.x/configuration.md

```diff
@@ -104,13 +107,138 @@ $ export KONG_LOG_LEVEL=error
+
+For example, if you the following line to your `kong.conf` file:
+
+```
+nginx_proxy_add_header=Header-Name header-value
+```
+
+This will add the following directive to the proxy `server` block of Kong's
+Nginx configuration:
+
+```
+    add_header Header-Name header-value;
+```
+
+Like any other entry in `kong.conf`, these directives can also be specified
+using [environment variables](#environment-variables) as shown above. For
+example, if you declare an environment variable like this:
+
+```
+$ export KONG_NGINX_HTTP_LUA_SHARED_DICT="custom_cache 64k"
+```
+
+This will result in the following Nginx directive being added to the `http`
+block:
+
+```
+    lua_shared_dict custom_cache 64k;
+```
+
+As always, be mindful of your shell's quoting rules specifying values
+containing spaces.
+
+For more details on the Nginx configuration file structure and block
+directives, see https://nginx.org/en/docs/beginners_guide.html#conf_structure.
+
+For a list of Nginx directives, see https://nginx.org/en/docs/dirindex.html.
+Note however that some directives are dependent of specific Nginx modules,
+some of which may not be included with the official builds of Kong.
+
+#### Including files via injected Nginx directives
+
+For more complex configuration scenarios, such as adding entire new
+`server` blocks, you can use the method described above to inject an
+`include` directive to the Nginx configuration, pointing to a file
+containing your additional Nginx settings.
+
+For example, if you create a file called `my-server.kong.conf` with
+the following contents:
+
+```
+# custom server
+server {
+  listen 2112;
+  location / {
+    # ...more settings...
+    return 200;
+  }
+}
+```
+
+You can make the Kong node serve this port by adding the following
+entry to your `kong.conf` file:
+
+```
+nginx_http_include /path/to/your/my-server.kong.conf
+```
+
+Or, alternatively, configuring it via an environment variable:
+
+```
+$ export KONG_NGINX_HTTP_INCLUDE="/path/to/your/my-server.kong.conf"
+```
+
+Now, when you start Kong the `server` section from that file will be added to
+that file, meaning that the custom server defined in it will be responding,
+alongside the regular Kong ports:
+
+```
+$ curl -I http://127.0.0.1:2112
+HTTP/1.1 200 OK
+...
+```
+
+Note that if you use a relative path in an `nginx_http_include` property, that
+path will be interpreted relative to the value of the `prefix` property of
+your `kong.conf` file (or the value of the `-p` flag of `kong start` if you
+used it to override the prefix when starting Kong).
+
+### Custom Nginx templates & embedding Kong
+
+For the vast majority of use-cases, using the Nginx directive injection system
+explained above should be sufficient for customizing the behavior of Kong's
+Nginx instance. This way, you can manage the configuration and tuning of
+your Kong node from a single `kong.conf` file (and optionally your own
+included files), without having to deal with Kong's configuration template.
```

Comment:
with "custom Nginx configuration templates" (Kong's configuration is simple enough and we should try avoiding possible confusion...)